test/unittest/blas1_rotg_test.cpp

```diff
@@ -0,0 +1,80 @@
+/***************************************************************************
+ *
+ *  @license
+ *  Copyright (C) 2016 Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  SYCL-BLAS: BLAS implementation using SYCL
+ *
+ *  @filename blas1_rotg_test.cpp
+ *
+ **************************************************************************/
+
+#include "blas1_test.hpp"
+
+typedef ::testing::Types<blas1_test_args<float>, blas1_test_args<double> >
+    BlasTypes;
+
+TYPED_TEST_CASE(BLAS1_Test, BlasTypes);
+
+REGISTER_SIZE(2, rot_test)
+REGISTER_STRD(1, rot_test)
+REGISTER_PREC(float, 1e-4, rot_test)
+REGISTER_PREC(double, 1e-7, rot_test)
+
+TYPED_TEST(BLAS1_Test, rot_test) {
+  using ScalarT = typename TypeParam::scalar_t;
+  using ExecutorType = typename TypeParam::executor_t;
+  using TestClass = BLAS1_Test<TypeParam>;
+  using test = class rot_test;
+
+  size_t size = TestClass::template test_size<test>();
+  size_t strd = TestClass::template test_strd<test>();
+  ScalarT prec = TestClass::template test_prec<test>();
+
+  std::vector<ScalarT> vX(size);
+  std::vector<ScalarT> vY(size);
+  TestClass::set_rand(vX, size);
+  TestClass::set_rand(vY, size);
+
+  for (auto &d : cl::sycl::device::get_devices()) {
+    std::vector<ScalarT> vZ(size, 0);
+    std::vector<ScalarT> vT(size, 0);
+    ScalarT _cos, _sin;
+
+    for (size_t i = 0; i < size; i += strd) {
+      ScalarT x = vX[i], y = vY[i];
+      _rotg(x, y, _cos, _sin);
```

Comment:
_rotg only computes the scalars which are required by _rot.
In theory, they can be computed where you prefer